backend/src/api/routes/storeController.js

```diff
@@ -0,0 +1,107 @@
+const express = require('express');
+const route = express.Router();
+const storeService = require("../../service/storeService")
+const Logger = require('../../service/loggerService')
+
+module.exports = (app) => {
+  const logger = new Logger('app')
+  app.use('/stores', route);
+
+/**
+ * @api {get} /stores/test Prueba de conexion al API
+ * @apiName Test
+ * @apiGroup Stores
+ *
+ * @apiSuccess {String} 'First Store'.
+ */
+  route.get('/test', (req, res) => {
+    return res.json({ store: "First Store" });
+  });
+
+
+  /**
+ * @api {post} /stores/getClosest Busca la tienda mas cercana
+ * @apiName GeTClosest
+ * @apiGroup Stores
+ * 
+ * @apiParam {Position} posicion en formato {"latitude": {Number}, "longitude": {Number}} 
+ *
+ * @apiSuccess {Store} {
+        storeId: store.code,
+        storeName: store.name,
+        isOpen: true,
+        coordinates: { latitude: store.latitude, longitude: store.longitude },
+        nextDeliveryTime: store.nextDeliveryTime,
+        status: "OK"
+      }.
+ */
+  route.post('/getClosest', async (req, res) => {
+    const location = req.body
+    logger.info(`/getClosest request recibido ${location.latitude}, ${location.longitude}`)
+    const store = await storeService.getClosest(location)
+    if (store) {
+      logger.info(`/getClosest response tienda ${store.code} para request ${location.latitude}, ${location.longitude}`)
+
+      return res.json({
+        storeId: store.code,
+        storeName: store.name,
+        isOpen: true,
+        coordinates: { latitude: store.latitude, longitude: store.longitude },
+        nextDeliveryTime: store.nextDeliveryTime,
+        status: "OK"
+      });
+    } else {
+      logger.error(`/getClosest no se encontraron tiendas para las coordenadas ${location.latitude}, ${location.longitude}`)
+      return res.json({
+        status: "NO_RESULT"
+      })
+    }
+
+  });
+
+  /**
+ * @api {post} /stores/getNearby Busca las tiendas mas cercana
+ * @apiName GetNearby
+ * @apiGroup Stores
+ * 
+ * @apiParam {Position} posicion en formato {"latitude": {Number}, "longitude": {Number}} 
+ *
+ * @apiSuccess {Store[]} {
+        storeId: store.code,
+        storeName: store.name,
+        isOpen: true,
+        coordinates: { latitude: store.latitude, longitude: store.longitude },
+        nextDeliveryTime: store.nextDeliveryTime,
+        status: "OK"
+      }.
+ */
+  route.post('/getNearby', async (req, res) => {
```

Comment:
getNearby returns all stores within an 8-mile radius, getClosest only the closest one, I wanted to use getNearby to show the user all the stores nearby, but in the end, because of time, and since it wasn't part of the test, I decided not to use it in the frontend, I removed it from the api

Reply:
Got it, thanks